yt/data_objects/data_containers.py

```diff
@@ -257,15 +257,23 @@ def __getitem__(self, key):
         # hanging off the dataset to define this unit object.
         # Note that this is less succinct so that we can account for the case
         # when there are, for example, no elements in the object.
-        rv = self.field_data.get(f, None)
-        if rv is None:
+        try:
+            rv = self.field_data[f]
+        except KeyError:
             if isinstance(f, tuple):
                 fi = self.ds._get_field_info(*f)
             elif isinstance(f, bytes):
                 fi = self.ds._get_field_info("unknown", f)
             rv = self.ds.arr(self.field_data[key], fi.units)
         return rv
 
+    def _ipython_key_completions_(self):
+        # to keep in sync with yt.data_objects.region_expression.RegionExpression
+        keys = self.ds.field_list + self.ds.derived_field_list
+        ftypes = list(set(k[0] for k in keys))
+        fnames = list(set(k[1] for k in keys))
+        return ftypes + fnames
```

Comment:
If https://github.com/ipython/ipython/pull/12395 is merged, then this could be rewritten simply as 
```suggestion
        return self.ds.field_list + self.ds.derived_field_list
```
and same below for `region_expression`.